### PR TITLE
Add `roleTooltipRequiresDescribedby` audit rule.

### PR DESCRIPTION
**Pull request by kennyw12**
#231 
http://www.w3.org/TR/2011/CR-wai-aria-20110118/roles#tooltip

### CODE REVIEW

**Comment by kennyw12** [comment]
For the Wiki:

## AX_TOOLTIP_01

#### Good:
```html
<!-- Good: each tooltip is referenced by a aria-describedby on a tooltip trigger -->
<p>Check out 
  <span aria-describedby='tp1'>
    this tooltip!
  </span>
</p>
<div role='tooltip' id='tp1'>This is a tooltip</div>
```

#### Bad:
```html
<!-- Bad: a tooltip is not referenced by a aria-describedby on a tooltip trigger -->
<p>Check out 
  <span>
    this terrible tooltip!
  </span>
</p>
<div role='tooltip' id='tp1'>This is a tooltip</div>
```

[WAI-ARIA tooltips:](http://www.w3.org/TR/2011/CR-wai-aria-20110118/roles#tooltip)
Elements with the role tooltip should be referenced by using aria-describedby by the time the tooltip is displayed.

File: src/audits/RoleTooltipRequiresDescribedBy.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * 'Elements with role=tooltip should have at least one other element with aria-describedby referring to them.'
+ */
+axs.AuditRules.addRule({
+    name: 'roleTooltipRequiresDescribedby',
+    heading: 'Elements with role=tooltip should have a corresponding element with aria-describedby',
+    url: 'https://github.com/GoogleChrome/accessibility-developer-tools/wiki/Audit-Rules#ax_aria_02',
+    severity: axs.constants.Severity.SEVERE,
+    relevantElementMatcher: function(element) {
+        return axs.browserUtils.matchSelector(element, '[role=tooltip]');
```
Comment (alice) [inline src/audits/RoleTooltipRequiresDescribedBy.js]: I would check for the element being visible at this point as well - as your (very nice) wiki text points out, the advice concerns a tooltip which is *displayed*.

File: test/audits/role-tooltip-requires-described-by-test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    fixture.appendChild(tooltip);
+    fixture.appendChild(trigger);
+    tooltip.setAttribute('role', 'tooltip');
+    tooltip.setAttribute('id', 'tooltip1');
+    trigger.setAttribute('aria-describedby', 'tooltip2');
+    deepEqual(
+        axs.AuditRules.getRule('roleTooltipRequiresDescribedby').run({ scope: fixture }),
+        { elements: [tooltip], result: axs.constants.AuditResult.FAIL }
+    );
+});
```
Comment (alice) [inline test/audits/role-tooltip-requires-described-by-test.js]: newline :)

File: test/audits/role-tooltip-requires-described-by-test.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    fixture.appendChild(tooltip);
+    tooltip.setAttribute('role', 'tooltip');
+    tooltip.setAttribute('id', 'tooltip1');
+    deepEqual(
+        axs.AuditRules.getRule('roleTooltipRequiresDescribedby').run({ scope: fixture }),
+        { elements: [tooltip], result: axs.constants.AuditResult.FAIL }
+    );
+});
+
+test('role tooltip without a corresponding aria-describedby should fail', function() {
```
Comment (alice) [inline test/audits/role-tooltip-requires-described-by-test.js]: nice test!

**Comment by kennyw12** [comment]
@alice 

Updated with fixes :) Thank you!

File: Changelog.md [hunk 4]
```diff
@@ -3,6 +3,7 @@
 ### New rules
 
 * A label element may not have labelable descendants other than its labeled control (`src/audits/MultipleLabelableElementsPerLabel.js`)
+* A tooltip element should have an aria-describedby referring to it (`src/audits/RoleTooltipRequiresDescribedBy.js`).
```
Comment (alice) [inline Changelog.md]: This will need to be pulled up to the top of the Changelog.

**Comment by alice** [comment]
Mostly looks good but I'm not sure it's up to date with the latest changes?

**Comment by kennyw12** [comment]
Sorry. it should be good now.

**Comment by alice** [comment]
:+1:

**Pull request merged**